test/libsolidity/smtCheckerTests/types/address_call.sol

```diff
@@ -20,8 +20,8 @@ contract C
 // ----
 // Warning 2072: (224-240): Unused local variable.
 // Warning 4588: (244-256): Assertion checker does not yet implement this type of function call.
-// Warning 6328: (260-275): CHC: Assertion violation happens here.
-// Warning 6328: (279-293): CHC: Assertion violation happens here.
-// Warning 6328: (297-316): CHC: Assertion violation happens here.
-// Warning 6328: (320-344): CHC: Assertion violation happens here.
+// Warning 6328: (260-275): CHC: Assertion violation happens here.\nCounterexample:\nx = 0\na = 0\n\nTransaction trace:\nC.constructor()\nState: x = 0\nC.f(0, data)
+// Warning 6328: (279-293): CHC: Assertion violation happens here.\nCounterexample:\nx = 1\na = 0\n\nTransaction trace:\nC.constructor()\nState: x = 0\nC.f(0, data)
+// Warning 6328: (297-316): CHC: Assertion violation happens here.\nCounterexample:\nx = 1\na = 0\n\nTransaction trace:\nC.constructor()\nState: x = 0\nC.f(0, data)
+// Warning 6328: (320-344): CHC: Assertion violation happens here.\nCounterexample:\nx = 1\na = 0\n\nTransaction trace:\nC.constructor()\nState: x = 0\nC.f(0, data)
```

Comment:
Not sure why we get counterexamples after the change.
Looking at the produced CEXs, I was wondering if other external call can modify state variables even if the contract does not expose any methods to modify them directly.

Reply:
Not with `call`. State variables can only be modified by the contract itself or by `delegatecall`.